### PR TITLE
Add robot file for support-notifications subscription

### PR DESCRIPTION
**Pull request by cherrycl**
Add robot file for support-notifications subscription

#108
Signed-off-by: Cherry Wang <cherry@iotechsys.com>

### CODE REVIEW

File: TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Negative.robot [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+
+ErrSubscriptionGET002 - Query all subscriptions with non-int value on limit
+    When Query All Subscriptions With Invalid Limit
+    Then Should Return Status Code "400"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+
+# /subscription/category/{category}
+ErrSubscriptionGET003 - Query subscriptions with non-existed category
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Negative.robot:29]: Shouldn't this case return 200 and empty array?

File: TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Negative.robot [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+
+ErrSubscriptionGET005 - Query subscriptions with specified category and non-int value on limit
+    When Query Subscriptions With Specified Category And Invalid Limit
+    Then Should Return Status Code "400"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+
+# /subscription/label/{label}
+ErrSubscriptionGET006 - Query subscriptions with non-existed label
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Negative.robot:48]: Shouldn't this case return 200 and empty array?

File: TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Negative.robot [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+
+ErrSubscriptionGET008 - Query subscriptions with specified label and non-int value on limit
+    When Query Subscriptions With Specified Label And Invalid Limit
+    Then Should Return Status Code "400"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+
+# /subscription/receiver/{receiver}
+ErrSubscriptionGET009 - Query subscriptions with non-existed receiver
```
Comment (jinlinGuan) [inline TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Negative.robot:67]: Shouldn't this case return 200 and empty array?

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

File: TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Positive.robot [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+# /subscription/all
+SubscriptionGET001 - Query all subscriptions that subscriptions are less then 20
+    Given Create Multiple Subscriptions
+    When Query All Subscriptions
+    Then Should Return Status Code "200"
+    And Should Return Content-Type "application/json"
+    And Response Time Should Be Less Than "${default_response_time_threshold}"ms
+    And Returned Subscription Records Should Be The Same As Created
+
+SubscriptionGET002 - Query all subscriptions that subscriptions more less then 20
```
[the remark this answers is not in the document]
Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Positive.robot]: Updated.

File: TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Negative.robot (hunk 1 above)
Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Negative.robot:29]: Confirmed with @FelixTing Query APIs with non-existed data should return 404.

Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Negative.robot]: Removed.

File: TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Negative.robot (hunk 3 above)
Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Negative.robot]: Removed.

File: TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Negative.robot (hunk 2 above)
Comment (cherrycl) [reply TAF/testScenarios/functionalTest/V2-API/support-notifications/subscription/GET-Negative.robot]: Removed.

**Review by jinlinGuan — APPROVED** [review]
LGTM

**Pull request merged**